### PR TITLE
fix up 3.11 glide deps

### PR DESCRIPTION
**Pull request by bparees**
fixes https://bugzilla.redhat.com/show_bug.cgi?id=1670551

effectively a cherry-pick of https://github.com/openshift/origin/pull/21533

### CODE REVIEW

**Comment by bparees** [comment]
@deads2k @mfojtik @soltysh I cleaned up a few things but this is currently failing to build due to some issues in kube/cli:

```
origin (mirror311)$ make build
hack/build-go.sh  
++ Building go targets for linux/amd64: cmd/hypershift cmd/openshift cmd/oc cmd/oadm cmd/template-service-broker cmd/openshift-node-config vendor/k8s.io/kubernetes/cmd/hyperkube cmd/sdn-cni-plugin vendor/github.com/containernetworking/plugins/plugins/ipam/host-local vendor/github.com/containernetworking/plugins/plugins/main/loopback
# github.com/openshift/origin/pkg/cmd/server/origin
pkg/cmd/server/origin/master.go:119:2: undefined: "github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/registry/core/rest".LegacyStorageMutatorFn
# github.com/openshift/origin/pkg/oc/cli
pkg/oc/cli/cli.go:347:3: undefined: util.DefaultPrintingScheme
pkg/oc/cli/cli.go:368:3: undefined: util.DefaultPrintingScheme
pullr	[ERROR] PID 1245: hack/lib/build/binaries.sh:236: `GOOS=${platform%/*} GOARCH=${platform##*/} go install -pkgdir "${pkgdir}/${platform}" -tags "${OS_GOFLAGS_TAGS-} ${!platform_gotags_envvar:-}" -ldflags="${local_ldflags}" "${goflags[@]:+${goflags[@]}}" -gcflags "${gogcflags}" "${nonstatics[@]}"` exited with status 2.
[INFO] 		Stack Trace: 
[INFO] 		  1: hack/lib/build/binaries.sh:236: `GOOS=${platform%/*} GOARCH=${platform##*/} go install -pkgdir "${pkgdir}/${platform}" -tags "${OS_GOFLAGS_TAGS-} ${!platform_gotags_envvar:-}" -ldflags="${local_ldflags}" "${goflags[@]:+${goflags[@]}}" -gcflags "${gogcflags}" "${nonstatics[@]}"`
[INFO] 		  2: hack/lib/build/binaries.sh:156: os::build::internal::build_binaries
[INFO] 		  3: hack/build-go.sh:25: os::build::build_binaries
[INFO]   Exiting with code 2.
[ERROR] PID 1184: hack/lib/build/binaries.sh:150: `( os::build::internal::build_binaries "${binaries[@]+"${binaries[@]}"}" )` exited with status 2.
[INFO] 		Stack Trace: 
[INFO] 		  1: hack/lib/build/binaries.sh:150: `( os::build::internal::build_binaries "${binaries[@]+"${binaries[@]}"}" )`
[INFO] 		  2: hack/build-go.sh:25: os::build::build_binaries
[INFO]   Exiting with code 2.
[ERROR] hack/build-go.sh exited with code 2 after 00h 00m 17s
make: *** [Makefile:43: build] Error 2

```

This means we can't currently bump origin-3.11 dependencies... can you take a look at the remaining failures?

**Comment by bparees** [comment]
(I'm happy to pin 3.11 deps to the kubernetes level that it was already using and working with, but i assume you'd prefer to fix up the kubernetes fork or the origin code).

**Comment by bparees** [comment]
Note that I added https://github.com/openshift/origin/pull/21887/commits/3782c5a88c5263362e70028e5ca731250b09e309 to get this compiling so i can make progress.  Again i'm happy to merge w/ that change, but i assume you don't want it going in.

**Comment by soltysh** [comment]
> Note that I added [3782c5a](https://github.com/openshift/origin/commit/3782c5a88c5263362e70028e5ca731250b09e309) to get this compiling so i can make progress. Again i'm happy to merge w/ that change, but i assume you don't want it going in.

Yeah, we'll fix the k8s fork.

**Comment by bparees** [comment]
> Yeah, we'll fix the k8s fork.

@soltysh thanks, ping me back on this PR when you've done that and i'll re-bump the deps.

**Comment by soltysh** [comment]
> @soltysh thanks, ping me back on this PR when you've done that and i'll re-bump the deps.

@bparees I've reset-ed the branches for 3.11 down the when the rebase landed. We need to wait for the publishing bot to pick up and catch up with the upstream commits. IIRC the bot runs every 4-6 hours. You can look at the commits in the forks.

**Comment by bparees** [comment]
@soltysh looks like new commits went in today:
https://github.com/openshift/kubernetes/commits/origin-3.11-kubernetes-1.11.1

but after updating deps things are still failing w/ the same compile problem.

**Comment by bparees** [comment]
/assign @smarterclayton 

@smarterclayton this is intended to fix the oc image mirror command in 3.11 (same as you did for 4.0).

File: glide.lock [hunk 1]
```diff
@@ -1859,7 +1860,7 @@ imports:
   - pkg/util/proto/testing
   - pkg/util/proto/validation
 - name: k8s.io/kubernetes
-  version: 91becade863adc3e9a01585d027896fd1f0d12a7
+  version: 763e9251f56c67913afb352101f541107b10448c
```
Comment (soltysh) [inline glide.lock]: It looks like you caught the publishing bot half way through, I'm just checking our fork and last commit there points to https://github.com/openshift/kubernetes/commit/7cd5a97e7f44e20b477a9fc42b85cd66c6c10d7a mind double checking?

Comment (bparees) [reply glide.lock]: i just reran hack/update-deps and it didn't pick up any changes, even after a glide cc.

Comment (bparees) [reply glide.lock]: So something else is pinning us to the earlier commit, presumably.

Comment (bparees) [reply glide.lock]: @soltysh @deads2k glide reports:
```
[INFO]  Keeping k8s.io/kubernetes origin-3.11-kubernetes-1.11.1
```

so i do not know why we're not getting the latest commit from that branch.  It's not my primary motivation in this PR, so unless one of you has a suggestion i'd like to merge this as is and you can chase the kube dep level as a follow up.

Comment (soltysh) [reply glide.lock]: Agreed, I'll straighten this up once your PR merges.

**Pull request merged**